delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -450,7 +450,20 @@ def _request_access_token(cls, delivery_carrier):
             },
             timeout=60,
         )
-        return response.json()
+
+        response.raise_for_status()
+
+        try:
+            json_response = response.json()
+        except requests.exceptions.JSONDecodeError as exc:
+            if not json_response:
+                raise exceptions.UserError(
+                    _("API response without content - no access token received.")
+                ) from exc
+            raise
```

Comment:
```suggestion
        try:
            json_response = response.json()
            response.raise_for_status()
        except requests.exceptions.JSONDecodeError as exc:
            raise exceptions.UserError(
                _("API response without content - no access token received.")
            ) from exc
        except requests.exceptions.HTTPError as exc:
            raise exceptions.UserError(
                _("Raise the message you think is better here")
            ) from exc
```
This is supposed to raise an exception when response code isn't 200 see the [documentation](https://3.python-requests.org/user/quickstart/#response-status-codes)

Shouldn't this be in the try/except clause?

Reply:
> This is supposed to raise an exception when response code isn't 200 see the [documentation](https://3.python-requests.org/user/quickstart/#response-status-codes)

No, it will not raise if code is a 2xx. Empty content is a 204 and `raise_for_status` will not raise anything in that case.

Also `raise_for_status` needs to happen before `response.json`, because if HTTP code is 4xx or 5xx, you want the corresponding error to be raised before attempting to convvert the response to json.

Reply:
Yeah, 2XX is what I meant.
Anyways, this needed to happen in the `try` block.